src/dataunion/DataUnion.ts

```diff
@@ -77,6 +77,22 @@ function getMessageHashes(tr: ContractReceipt): AmbMessageHash[] {
     return hashes
 }
 
+async function waitForTx(tx: ContractTransaction): Promise<ContractReceipt> {
+    return tx.wait().catch((e) => {
+        log('Got error: %o', e)
```

Comment:
Does the error message include information about which transaction failed, wondering if something like `tx.hash` should be printed here too.

Reply:
good point, might as well print the tx too